pkg/lib/common.ts

```diff
@@ -26,32 +28,32 @@ export function in_array(array, val) {
     return false;
 }
 
-export function is_function(x) {
+export function is_function(x: unknown): x is (...args: unknown[]) => unknown {
     return typeof x === 'function';
 }
 
-export function is_object(x) {
+export function is_object(x: unknown): x is object {
     return x !== null && typeof x === 'object';
 }
 
-export function is_plain_object(x) {
+export function is_plain_object(x: unknown): boolean {
```

Comment:
We really should have added documentation to this a decade ago.. I wouldn't know now when to use `is_object` and `is_plain_object` and which objects aren't plain.. :sob:

Reply:
See also `is_json_dict()` in `utils.tsx`...

Reply:
Not much actionable, this was mostly a rant. But it factors into the "let's not make this publicly accessible pkg/lib API" concern below.